### PR TITLE
Logging service extra

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

**Comment by Therzok** [comment]
The main reason for this change is that usually logs sent by Subversion users do not have the native error code attached, so logs will now have it. Probably a line like "Exception Data:" can be prepended to the part where we log the extra data.

File: main/src/core/MonoDevelop.Core/MonoDevelop.Core/LoggingService.cs [hunk 1]
```diff
@@ -492,20 +493,35 @@ public static void LogFatalError (string messageFormat, params object[] args)
...
-		
+
+		static string FormatExceptionText (string message, Exception ex)
+		{
+			var exceptionText = new StringBuilder ();
+			exceptionText.Append (message);
+			if (ex != null) {
+				exceptionText.AppendLine ();
+				exceptionText.Append (ex);
+				foreach (DictionaryEntry item in ex.Data) {
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Core/LoggingService.cs]: Should I prepend "Exception Data:" as a line here?

Comment (slluis) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Core/LoggingService.cs]: Yes, although only if ex.Data is not empty. Also, if ex is null it should return message, no need to create the StringBuilder in this case.

Comment (Therzok) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Core/LoggingService.cs]: Sure, I'll get on to these changes.

**Comment by slluis** [comment]
:+1:

**Comment by Therzok** [comment]
Merged manually.

**Pull request closed without merging**